### PR TITLE
win_domain and win_domain_controller - add support for logpath parameter

### LINKED ISSUE #59348: win_domain / win_domain_controller - log path missing
##### SUMMARY

The equivelent powershell command to win_domain includes the option to specify log path:
```
Install-ADDSForest `
-DatabasePath "D:\Windows\NTDS" `
-SysvolPath "D:\Windows\SYSVOL" `
-LogPath "E:\Windows\NTDS"
```
This has been missed in the win_domain module but SYSVOL and Database path are included, this is not consistent.
In addition to this, the win_domain_controller does not include it as a parameter in docs but it is in the examples! but also has no effect.

Should be an easy fix to make, otherwise please fix the example in docs. Would be greatly appreciated. The only way to change this is by rebooting into restore mode while AD if offline.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
win_domain
win_domain_controller

##### ANSIBLE VERSION

```
ansible 2.8.2
  config file = /Users/redacted/projects/project/ansible/ansible.cfg
  configured module search path = ['/Users/redacted/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.8.2/libexec/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.4 (default, Jul  9 2019, 18:13:23) [Clang 10.0.1 (clang-1001.0.46.4)]
```

##### CONFIGURATION
For example, the config below is ignored and the logs are in the C:\ drive
```
    win_domain:
      create_dns_delegation: no
      dns_domain_name: '{{ prod_domain_name }}'
      domain_netbios_name: '{{ prod_netbios }}'
      domain_mode: WinThreshold
      forest_mode: WinThreshold
      safe_mode_password: '{{ prod_dsrm_pass }}'
      database_path: D:\Windows\NTDS
      sysvol_path: D:\Windows\SYSVOL
      log_path: E:\Windows\NTDS
```

##### OS / ENVIRONMENT
Non OS Specific, Tested on Server 2019  and 2016

##### STEPS TO REPRODUCE
n/a

##### EXPECTED RESULTS
Logs go to E:\Windows\NTDS\ etc...

##### ACTUAL RESULTS
Logs go to %systemroot%\NTDS\

### PR DESCRIPTION
**Pull request by ShachafGoldstein**
##### SUMMARY
Fixes: #59348

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
win_domain.ps1
win_domain_controller.ps1

### CODE REVIEW

File: lib/ansible/modules/windows/win_domain_controller.ps1 [hunk 1]
```diff
@@ -257,7 +258,7 @@ Try {
             $local_admin_secure = $local_admin_password | ConvertTo-SecureString -AsPlainText -Force
 
             Write-DebugLog "Uninstalling domain controller..."
-            Uninstall-ADDSDomainController -NoRebootOnCompletion -LocalAdministratorPassword $local_admin_secure -Credential $domain_admin_cred
+            Uninstall-ADDSDomainController -NoRebootOnCompletion -LocalAdministratorPassword $local_admin_secure -Credential $domain_admin_cred -LogPath $log_path
```
Comment (agowa) [inline lib/ansible/modules/windows/win_domain_controller.ps1]: will this work? log_path is not necessarily defined. Shouldn't it be something like:
```suggestion
            Uninstall-ADDSDomainController -NoRebootOnCompletion -LocalAdministratorPassword $local_admin_secure -Credential $domain_admin_cred -LogPath:$log_path
```

File: lib/ansible/modules/windows/win_domain_controller.ps1 [hunk 2]
```diff
@@ -196,6 +196,7 @@ Try {
                 $install_params = @{
                     DomainName = $dns_domain_name
                     Credential = $domain_admin_cred
+                    LogPath = $log_path
```
Comment (agowa) [inline lib/ansible/modules/windows/win_domain_controller.ps1:199]: should it even be added to $install_params if $log_path is null?

**Review by agowa — COMMENTED** [review]
:+1:

File: lib/ansible/modules/windows/win_domain_controller.ps1 (hunk 2 above)
Comment (ShachafGoldstein) [reply lib/ansible/modules/windows/win_domain_controller.ps1:199]: if the param is `null` it ignores it

File: lib/ansible/modules/windows/win_domain_controller.ps1 (hunk 1 above)
Comment (ShachafGoldstein) [reply lib/ansible/modules/windows/win_domain_controller.ps1]: that parameter isn't relevant for this cmdlet apparently

File: lib/ansible/modules/windows/win_domain_controller.ps1 (hunk 2 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_domain_controller.ps1:199]: I would still only add it if it was defined like we do with the others, who knows if that behaviour ever changes.

**Review by jborean93 — COMMENTED** [review]
While I'm ok with adding the `LogPath` parameter we have 1 problem with it in the `win_domain_controller` module. We currently use (and document in an example) a hidden option called `log_path` which is used to place debug log files generated by the module itself. This is a file and not a directory that seems to be expected by the Install cmdlet so we need to solve that issue.

The other problem is the wording the docs, you are saying it is the log file of the installation but I'm wondering where it does more than that considering you pass a directory and not a single filename.

**Comment by ShachafGoldstein** [comment]
> While I'm ok with adding the `LogPath` parameter we have 1 problem with it in the `win_domain_controller` module. We currently use (and document in an example) a hidden option called `log_path` which is used to place debug log files generated by the module itself. This is a file and not a directory that seems to be expected by the Install cmdlet so we need to solve that issue.
> 
> The other problem is the wording the docs, you are saying it is the log file of the installation but I'm wondering where it does more than that considering you pass a directory and not a single filename.

I am reading again the docs and it doesn't seem to be the installation log but an actual production log path.
i will comment on the original issue, lets put this PR on hold until I figure it out fully

**Comment by ShachafGoldstein** [comment]
Ok the original issue is a request to change this parameter of log path so it has nothing to do with the installation logs

**Comment by jborean93** [comment]
@ShachafGoldstein are you still wanting to continue with this PR, if not let's close it and revisit it at a later point in time.

**Comment by ShachafGoldstein** [comment]
> @ShachafGoldstein are you still wanting to continue with this PR, if not let's close it and revisit it at a later point in time.

yes

**Comment by jborean93** [comment]
This has been implemented in 2 separate PRs

* `win_domain`: https://github.com/ansible/ansible/pull/66956
* `win_domain_controller`: https://github.com/ansible/ansible/pull/67448

**Pull request closed without merging**